File: data_structures/stack/array_stack.js
class Stack {
    constructor() {
        this.stack = [];
    }

    push(element) {
        this.stack.push(element)
    }

    pop() {
        this.stack.pop()
    }

    peek() {
        if (isEmpty()) return;
        return this.stack[this.stack.length - 1];
    }

    clear() {
        this.stack = [];
    }

    display() {
        console.log(this.stack);
    }

    isEmpty() {
        return this.stack.length === 0;
    }
}

File: data_structures/stack/array_stack.rs
#[derive(Debug)]
struct Stack {
    stack: Vec<T>
}

impl Stack {
    fn new() -> Self {
        Stack {
            stack: Vec::new()
        }
    }

    fn push(self, element: T) {
        self.stack.push(element);
    }

    fn pop(self) -> <T> {
        return self.stack.pop();
    }

    fn peek(self) -> Option<T> {
        if (self.isEmpty()) return None;
        return Some(self.stack[self.stack.length - 1]);
    }

    fn clear(self) {
        self.stack = Vec::new();
    }

    fn display(self) {
        println!("{:?}", self.stack)
    }

    fn isEmpty(self) -> Boolean {
        return self.stack.len() == 0;
    }
}